src/pages/Signup/Signup.js

```diff
@@ -1,41 +1,121 @@
-import React from 'react'
-import { Link } from 'react-router-dom'
+import React, { useEffect, useReducer } from 'react'
+import { useDispatch } from 'react-redux'
+import { Link, useNavigate } from 'react-router-dom'
+import { signupFormReducer } from './signupFormReducer'
+import { signupUser } from 'store/feature/authSlice'
+import { useAuth } from 'hooks/selectors'
 const Signup = () => {
+    const dispatch = useDispatch();
+    const { user } = useAuth();
+    const navigate = useNavigate();
+    const [{ firstName, lastName, username, email, password, confirmPassword }, signupDispatch
+    ] = useReducer(signupFormReducer, { firstName: "", lastName: "", username: "", email: "", password: "", confirmPassword: "" })
+
+    const submitHandler = async (e, firstName, lastName, username, email, password, confirmPassword) => {
+        e.preventDefault();
+        dispatch(signupUser({ firstName, lastName, username, email, password, confirmPassword }));
+    }
```

Comment:
same here

Reply:
since `firstName, lastName, username, email, password, confirmPassword` these values are available in the function's outer scope, you need not pass these values as arguments to the function, the function would take these values from its outer scope